contracts/crypto-verify/src/contract.rs

```diff
@@ -0,0 +1,192 @@
+use sha2::{Digest, Sha256};
+
+use cosmwasm_std::{
+    attr, entry_point, to_binary, Binary, Deps, DepsMut, Env, MessageInfo, QueryResponse, Response,
+    StdResult,
+};
+
+use crate::msg::list_verifications;
+use crate::msg::{HandleMsg, InitMsg, ListVerificationsResponse, QueryMsg};
+
+pub const VERSION: &str = "crypto-verify-v1";
+
+#[entry_point]
+pub fn init(_deps: DepsMut, _env: Env, _info: MessageInfo, _msg: InitMsg) -> StdResult<Response> {
+    Ok(Response::default())
+}
+
+#[entry_point]
+pub fn handle(deps: DepsMut, _env: Env, _info: MessageInfo, msg: HandleMsg) -> StdResult<Response> {
+    match msg {
+        HandleMsg::VerifySignature {
+            message,
+            signature,
+            public_key,
+        } => handle_verify(deps, &message.0, &signature.0, &public_key.0),
+    }
+}
+
+pub fn handle_verify(
+    deps: DepsMut,
+    message: &[u8],
+    signature: &[u8],
+    public_key: &[u8],
+) -> StdResult<Response> {
+    // Hashing
+    let hash = Sha256::new().chain(message).finalize();
```

Comment:
Nitpick: `Sha256::digest(message)` does the same and is shorter.
https://docs.rs/sha2/0.9.3/sha2/trait.Digest.html#tymethod.digest